api/src/Entity/ScheduleEntry.php

```diff
@@ -2,58 +2,116 @@
+    #[Groups(['Default', 'scheduleEntry:update'])]
     public ?Period $period = null;
 
     /**
+     * The activity that will take place at the time defined by this schedule entry. Can not be changed
+     * once the schedule entry is created.
+     *
      * @internal Do not set the {@see Activity} directly on the ScheduleEntry. Instead use {@see Activity::addScheduleEntry()}
      * @ORM\ManyToOne(targetEntity="Activity", inversedBy="scheduleEntries")
      * @ORM\JoinColumn(nullable=false, onDelete="cascade")
      */
+    #[ApiProperty(example: '/activities/1a2b3c4d')]
+    #[Groups(['Default'])]
     public ?Activity $activity = null;
 
     /**
+     * The number of minutes that have passed since the start of the period when this schedule entry
+     * starts.
+     *
      * @var int minutes since period start
      * @ORM\Column(type="integer", nullable=false)
      */
+    #[Assert\GreaterThanOrEqual(value: 0)]
+    #[ApiProperty(example: 1440)]
+    #[Groups(['Default', 'scheduleEntry:update'])]
     public int $periodOffset = 0;
 
     /**
-     * @var int minutes
+     * The duration in minutes that this schedule entry will take.
+     *
      * @ORM\Column(type="integer", nullable=false)
      */
+    #[Assert\GreaterThan(value: 0)]
+    #[ApiProperty(example: 90)]
+    #[Groups(['Default', 'scheduleEntry:update'])]
     public int $length = 0;
 
     /**
+     * When rendering a period in a calendar view: Specifies how far offset the rendered calendar event
+     * should be from the left border of the day column, as a fractional amount of the width of the whole
+     * day. This is useful to arrange multiple overlapping schedule entries such that all of them are
+     * visible. Should be a decimal number between 0 and 1, and left+width should not exceed 1, but the
+     * API currently does not enforce this.
+     *
      * @ORM\Column(name="`left`", type="float", nullable=true)
      * --> left is a MariaDB keyword, therefore escaping for column name necessary
      */
-    public float $left = 0;
+    #[ApiProperty(default: 0, example: 0.6)]
+    #[Groups(['Default', 'scheduleEntry:update'])]
+    public ?float $left = 0;
 
     /**
+     * When rendering a period in a calendar view: Specifies how wide the rendered calendar event should
+     * be, as a fractional amount of the width of the whole day. This is useful to arrange multiple
+     * overlapping schedule entries such that all of them are visible. Should be a decimal number
+     * between 0 and 1, and left+width should not exceed 1, but the API currently does not enforce this.
+     *
      * @ORM\Column(type="float", nullable=true)
      */
-    public float $width = 1;
+    #[ApiProperty(example: 0.4)]
+    #[Groups(['Default', 'scheduleEntry:update'])]
+    public ?float $width = 1;
 
     #[ApiProperty(readable: false)]
     public function getCamp(): ?Camp {
-        return $this->period?->camp;
+        return $this->activity?->camp;
```

Comment:
Wieso dieser wechsel?

Reply:
Die Activity kann, im Gegensatz zur Periode, beim Bearbeiten eines ScheduleEntries nicht mehr geändert werden. Das erlaubt es mir, eine Benutzung des `AssertBelongsToSameCamp(compareToPrevious: true)` Hacks zu vermeiden. In meinem Kopf gehört ein ScheduleEntry zudem primär zu einer Activity, und erst zweitrangig zu einer Periode.

Ich könnte das aber zurückändern und die `AssertBelongsToSameCamp` Validierung von `period` auf `activity` verschieben, wenn du einen Grund dafür hast.

Reply:
Für mich ok. Man muss immer abwägen, ob man einen Fehler beim verschieben gleich korrigiert. Passt hier für mich so.